### ISSUE TITLE
feat: add support for continuous scanning (multiple barcodes in one go)

### ISSUE THREAD

**Issue by parnanzio** [capacitor-community/barcode-scanner#43]
Is there a way to scan multiple QR codes at once?

**Comment by tafelnl** [comment]
Do you mean multiple qr-codes at once? Or just continuous scanning without closing the camera inbetween?

**Comment by parnanzio** [comment]
Continuous scanning without closing the camera.

**Comment by tafelnl** [comment]
There is no way yet. But it is on the roadmap. Related issue #33

**Comment by iampapaisarkar** [comment]
yes you can, simply do this
![2021-06-06](https://user-images.githubusercontent.com/50805880/120972692-a3f42480-c78b-11eb-8b4d-11f282dde481.png)

call scan function after get result

**Comment by m-bagnarelli** [comment]
Hi, any updates on this? :). I really need it

**Comment by rulaman123** [comment]
@iampapaisarkar Yes, you can start the scanner again like that but it is ugly because the view will close and open again. I found a better way. Don't know if it's perfect but in my case it's working fine.

You have to edit the BarcodeScanner.java (capacitor-community\barcode-scanner\android\src\main\java\com\dutchconcepts\capacitor\barcodescanner\BarcodeScanner.java).

Go to the method "Override public void barcodeResult(BarcodeResult barcodeResult)" and delete
`destroy();`
Then the scanner view will not pause any more.

After that you can create a new method for scanning again.

```
private void scanAgain() {
    BarcodeCallback b = this;
    getActivity()
      .runOnUiThread(
        () -> {
            mBarcodeView.decodeSingle(b);
        }
      );
  }
```

After that create a new plugin method.

```
  @PluginMethod
  public void startScanAgain(PluginCall call) {
    saveCall(call);
    scanAgain();
  }
```

And the last step is to add the newly created method to the interface (capacitor-community\barcode-scanner\dist\esm\definitions.d.ts):
Go to the method "export interface BarcodeScannerPlugin" and add
`startScanAgain(): Promise<ScanResult>;`

And now you can start a new scan with "startScanAgain()".
I have done that by using a modal for my scan result and every time the model will dismiss the next scan is starting.
Dont forget to close the scanner, e.g. when leaving the site.

**Comment by ttruty** [comment]
@rulaman123 solution worked great for me. Thank you

**Comment by JanMisker** [comment]
@ttruty @rulaman123 check my PR https://github.com/capacitor-community/barcode-scanner/pull/93, solution for iOS and Android similar to how you did it.

**Comment by thegnuu** [comment]
PR #93 is merged and `2.1.0` with this feature will be released next week!
